### PR TITLE
Fix null error caused by certain elements called before they exist.

### PR DESCRIPTION
**Pull request by sjpadgett**
fix of javascript error on select datatype reported in pr https://github.com/openemr/openemr/pull/1152

### CODE REVIEW

**Comment by bradymiller** [comment]
Looks good and will bring into codebase. Thanks for the fix! -brady

**Comment by matrix-amiel** [comment]
@sjpadgett 
Sorry it's still not working..
There is another JS error.
image

File: interface/super/edit_layout.php [hunk 1]
```diff
@@ -2090,21 +2090,25 @@ function layoutLook(){
...
-    document.getElementById("fld["+lineitem+"][list_id]").value='';
-  }
-  else{
-    document.getElementById("fld["+lineitem+"][list_id]").style.display='';
-    document.getElementById("fld["+lineitem+"][contextName]").style.display='none';
-    document.getElementById("fld["+lineitem+"][list_id]").value='';
-  }
+function NationNotesContext(lineitem, val) {
+    // Check if function is needed.
+    if (!document.body.contains("fld[" + lineitem + "][contextName]") && !document.getElementById("fld[" + lineitem + "][list_id]")) {
```
Comment (sunsetsystems) [inline interface/super/edit_layout.php]: As Amiel points out this is broken (a string is not a node), but also you can simplify by using jQuery to hide, show and set values; it will just ignore unmatched selectors.

**Comment by sunsetsystems** [comment]
Can you explain why you think this is a timing error?

File: interface/super/edit_layout.php (hunk 1 above)
Comment (sjpadgett) [reply interface/super/edit_layout.php]: I didn't write this function and the conditional is only checking if the element exist. This fix is needed because when the form loads, php checks for certain data types and adds fields using contextName and list_id as element id's. Not every field in the form uses that id format however onchange event in datatype select always calls NationNoteContext even on datatypes that don't use the context or list_id id format.
Originally I was suspect the onchange function was being called before value was set after the event(proved not to be this case). I've seen it many times and better to set that event outside the element.
Also, if I'm modifying existing code, I try to stay with the same code style of original code(seems good manners:)). Trust me, I very much wanted to use JQuery here.
Summing up, mod should be fine and a DOM content node is not being access here but merely if any element containing those two id's exist. Btw, this behavior has there before we touched form. Why now, beats me...
I think there is another issue here because I tested this up and down...But, this is Monday.

Comment (sjpadgett) [reply interface/super/edit_layout.php]: oops, i see. i'll just rewrite with jquery. sorry, i'm becoming short sighted...
